workflow/workflow.go

```diff
@@ -62,12 +64,59 @@ func (w *Workflow) process(trigger service.TriggerType, instanceHash hash.Hash,
 	for _, service := range services {
 		for _, wf := range service.Workflows {
 			if wf.Trigger.Match(trigger, instanceHash, key, data) {
-				_, err := w.execution.Execute(wf.Task.InstanceHash, eventHash, executionHash, wf.Task.TaskKey, data, []string{})
-				if err != nil {
+				if err := w.triggerExecution(wf, exec, eventHash, data); err != nil {
 					w.ErrC <- err
-					continue
 				}
 			}
 		}
 	}
 }
+
+func (w *Workflow) processExecution(exec *execution.Execution) error {
+	if exec.WorkflowHash.IsZero() {
```

Comment:
What is the current state of workflow schema? Is workflow part of service or not? Why core needs workflow to execute single execution?

Reply:
Workflow is still part of the service but will certainly move into its own ressource and api soon.

> Why core needs workflow to execute single execution

That's the whole purpose of workflows to automatically trigger executions based on some definition. I'm not sure I really understand your question here